test/test_other.py

```diff
@@ -15805,10 +15805,18 @@ def test_rollup(self):
   def test_rlimit(self):
     self.do_other_test('test_rlimit.c', emcc_args=['-O1'])
 
-  def test_mainScriptUrlOrBlob(self):
+  @parameterized({
+    '': (False,),
+    'es6': (True,),
+  })
+  def test_mainScriptUrlOrBlob(self, es6):
     # Use `foo.js` instead of the current script name when creating new threads
+    if es6:
+      self.emcc_args += ['-sEXPORT_ES6', '--extern-post-js', test_file('modularize_post_js.js')]
+
     create_file('pre.js', 'Module = { mainScriptUrlOrBlob: "./foo.js" }')
-    self.run_process([EMCC, test_file('hello_world.c'), '-sEXIT_RUNTIME', '-sPROXY_TO_PTHREAD', '-pthread', '--pre-js=pre.js'])
+
+    self.run_process([EMCC, test_file('hello_world.c'), '-sEXIT_RUNTIME', '-sPROXY_TO_PTHREAD', '-pthread', '--pre-js=pre.js', '-o', 'a.out.js'])
```

Comment:
I think you can remove the `'-o', 'a.out.js'` but you maybe need to add `self.get_emcc_args()` the array ( sorry I forgot this test wasn't using a helper that did this already.

Reply:
Oh wait maybe you do still the explicit output name (`'-o', 'a.out.js'`) if `EXPORT_ES6` is added.

Reply:
I decided to keep it explicit just to be clear what file is being generated, since it’s important to the test that it has a particular name.

Reply:
I think you still need to add `self.get_emcc_args()` here in order to ensure that `-sEXPORT_ES6` is added.

Alternatively you should change the parameter `args` and do something like this:

```
@parameterized({
   '': ([],),
   'es6': ([-sEXPORT_ES6', '--extern-post-js', test_file('modularize_post_js.js')],),
})
def test_mainScriptUrlOrBlob(self, args):
    ...
     self.run_process([EMCC, test_file('hello_world.c'), '-sEXIT_RUNTIME', '-sPROXY_TO_PTHREAD', '-pthread', '--pre-js=pre.js', '-o', 'a.out.js'] + args)
```

Reply:
I like that better, thanks!